src/Middleware/HttpLogging/src/HttpLoggingInterceptorContext.cs

```diff
@@ -0,0 +1,122 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Numerics;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.HttpLogging;
+
+/// <summary>
+/// The context used for logging customization callbacks.
+/// </summary>
+public sealed class HttpLoggingInterceptorContext
+{
+    private HttpContext? _httpContext;
+
+    /// <summary>
+    /// The request context.
+    /// </summary>
+    public HttpContext HttpContext
+    {
+        get => _httpContext ?? throw new InvalidOperationException("HttpContext was not initialized");
+        // Public for 3rd party testing of interceptors.
+        // We'd make this a required constructor/init parameter but ObjectPool requires a parameterless constructor.
+        set => _httpContext = value ?? throw new ArgumentNullException(nameof(value));
+    }
+
+    /// <summary>
+    /// What parts of the request and response to log.
+    /// </summary>
+    public HttpLoggingFields LoggingFields { get; set; }
+
+    /// <summary>
+    /// Limits how much of the request body to log.
```

Comment:
```suggestion
    /// Gets or sets the maximum number of bytes of the request body to log.
```

Just to be explicit about the unit (bytes).